Client/Frontend/Browser/TabManager.swift

```diff
@@ -54,6 +54,8 @@ class TabManager : NSObject {
     private let defaultNewTabRequest: NSURLRequest
     private let navDelegate: TabManagerNavDelegate
     private var configuration: WKWebViewConfiguration
+    private var networkActivityCount = 0 // Counts the number of tabs with network activity
+    private var tabsNetworkActivity = [WKWebView: Bool]()
```

Comment:
I still need `tabsNetworkActivity` because `didStartProvisionalNavigation` gets called twice, plus the user can click links even if the page didn't finish loading, so I need to keep track of each tab's network activity. I think removing the `WKWebView` instance for each tab the user closes should fix the memory leak.